src/api/instance/instance.js

```diff
@@ -0,0 +1,23 @@
+// @flow
+import {type CredrankInput} from "../credrank";
+import {CredGraph} from "../../core/credrank/credGraph";
+import {Ledger} from "../../core/ledger/ledger";
+import {type DependenciesConfig} from "../dependenciesConfig";
+
+/**
+This interface represents a SourceCred Instance, which is simply a read/write
+interface for configurations and outputs of the main SourceCred API.
+ */
+export interface Instance {
```

Comment:
Optional, and can do later: any thought for separating out the writable
part of the interface? Following up on my suggestion above:

```javascript
interface Instance {
  /** Reads inputs required to run CredRank. */
  readCredrankInput(): Promise<CredrankInput>;
  /** Writes output after running CredRank. */
  writeCredrankOutput(o: CredrankOutput): Promise<void>;

  /** Reads a cred graph generated by a previous run of CredRank. */
  readCredGraph(): Promise<CredGraph>;
}
interface WritableInstance extends Instance {
  /** Writes output after running CredRank. */
  writeCredrankOutput(o: CredrankOutput): Promise<void>;
}
```

I suspect that this might be useful when you want to write an `Instance`
implementation that reads from GitHub without authentication, at which
point you won’t be able to implement the `write*` methods.

Reply:
Yeah, I also thought this would be useful in the future. Sure, it's pretty simple to add in.